index.js

```diff
@@ -150,15 +151,19 @@ class DHT extends EventEmitter {
 
     if (opts && opts.size && opts.size > 0) value = b4a.alloc(opts.size)
 
-    const req = this.io.createRequest({ id: null, host, port }, null, true, PING, null, value)
+    const req = this.io.createRequest({ id: null, host, port }, null, true, PING, null, value, opts && opts.session)
     return this._requestToPromise(req, opts)
   }
 
   request ({ token = null, command, target = null, value = null }, { host, port }, opts) {
-    const req = this.io.createRequest({ id: null, host, port }, token, false, command, target, value)
+    const req = this.io.createRequest({ id: null, host, port }, token, false, command, target, value, opts && opts.session)
```

Comment:
can we make session explciit in both of these? (opts && opts.session) || null

makes it easier to see bugs later